rocket_sqlite_oracle/src/main.rs

```diff
@@ -1,68 +1,77 @@
-use anyhow::{Context, Result};
+use anyhow::Result;
 use rocket::fairing::AdHoc;
 use rocket::serde::json::Json;
-use rocket::serde::Serialize;
-use rocket::{fairing, get, routes, tokio, Build, Rocket, State};
+use rocket::{get, routes, State};
 use rocket_db_pools::{sqlx, Connection, Database};
-use rocket_sqlite_oracle::kraken;
 use secp256k1::rand::thread_rng;
 use secp256k1::{PublicKey, SecretKey, Signature};
 use secp256k1_zkp::SECP256K1;
-use std::convert::{TryFrom, TryInto};
-use std::path::{Path, PathBuf};
+use std::convert::TryInto;
 use structopt::StructOpt;
-use tokio::io::AsyncWriteExt;
-use url::Url;
 
-use baru::oracle::Message;
+mod cli;
+mod db;
 
-#[derive(Database)]
-#[database("oracle_signatures")]
-struct ExampleDB(sqlx::SqlitePool);
-
-#[derive(Serialize)]
-struct USDTickerResponse {
-    timestamp: u64,
-    ask_price: u64,
-    bid_price: u64,
-    signature: Signature,
-}
-
-/// API endpoint definitions
 #[get("/")]
 fn oracle_pubkey(key: &State<SecretKey>) -> String {
     PublicKey::from_secret_key(SECP256K1, key).to_string()
 }
 
-#[get("/price")]
-async fn current_price(mut conn: Connection<ExampleDB>) -> Json<USDTickerResponse> {
+#[get("/ask/price")]
+async fn current_ask_price(mut conn: Connection<db::Signatures>) -> Json<db::USDTickerResponse> {
+    let row = sqlx::query!(
+        r#"
+        select
+            timestamp,
+            ask_price as price,
+            ask_signature as signature
+        from signatures
+        order by timestamp desc
+        limit 1;
+        "#,
+    )
+    .fetch_one(&mut *conn)
+    .await
+    .unwrap();
+    let res = db::USDTickerResponse {
+        timestamp: row.timestamp.try_into().unwrap(),
+        price: row.price.try_into().unwrap(),
+        signature: Signature::from_compact(&row.signature).unwrap(),
+    };
```

Comment:
This response is not DB specific but belongs semantically to the rocket endpoints. I don't think it should live in the `db` module.

Reply:
This was done based on your previous request...

Reply:
Oh, you must have misunderstood me then. What I meant was: Everything that is semantically related to the database should go in the DB module.

The struct that defines how our JSON response looks like has nothing to do with the database, it belongs to the endpoints that use it (which all live in `main.rs`).